other/check-vpa-configuration/check-vpa-configuration.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: check-matching-vpa
+  annotations:
+    policies.kyverno.io/title: Ensure Matching VerticalPodAutoscaler 
+    policies.kyverno.io/category: Availability
+    policies.kyverno.io/severity: medium
+    kyverno.io/kyverno-version: 1.11.4 
+    kyverno.io/kubernetes-version: "1.27"     
+    policies.kyverno.io/subject: Deployment, StatefulSet, ReplicaSet, DaemonSet
+    policies.kyverno.io/description: >-
+      This policy enforces that Deployments, StatefulSets, Replicasets, and DaemonSets have 
+      a corresponding VerticalPodAutoscaler to ensure potential automatic resource adjustments. 
+spec:
+  validationFailureAction: Audit
+  background: false
+  rules:
+    - name: require-vpa
+      match:
+        any:
+        - resources:
+            kinds:
+              - Deployment
+              - StatefulSet
+              - ReplicaSet
+              - DaemonSet
+      validate:
+        message: >-
+          Workload '{{request.object.kind}}/{{request.object.metadata.name}}' 
+          requires a matching VerticalPodAutoscaler (VPA) in the 
+          '{{request.object.metadata.namespace}}' namespace.
+      context:
+      - name: metricsServer
+        apiCall:
+          urlPath: "/apis/metrics.k8s.io/v1beta1/namespaces/{{ request.namespace }}/pods"
+          jmesPath:  
+      preconditions:
+        any:
+        - key: "{{ metricsServer }}"
+          operator: GreaterThan
+          value: 0 
+        foreach:
+          - list: "request.object.kind"
+            context:
+              - name: vpas
+                apiCall:
+                  urlPath: "/apis/autoscaling.k8s.io/v1/namespaces/{{request.object.metadata.namespace}}/verticalpodautoscalers"
+                  jmesPath: "items[?spec.targetRef.kind=='{{ request.object.kind }}'].spec.targetRef.name"
```

Comment:
While it doesn't hurt, I don't think it's really necessary to do so. The API version for all these resources is going to be `apps/v1` and that has been around for quite a long time at this point. Since those are going to be the vast majority, if not all, of the VPAs configured, it really doesn't help narrow down the matches.